### PR TITLE
resolve #21: Name changes for Graph, Node, and Node attributes

### LINKED ISSUE #21: Name changes for Graph, Node, and Node attributes
To implement the following changes to the current naming convention:
`Graph` to `Project`
`Node` to `Task`
`Node.threads` to `Task.subtasks`
`Node.dependencies` to `Task.blockers`
`Node.weight` to `Task.priority`

### PR DESCRIPTION
**Pull request by capsulecorplab**
closes #21 

Additions:
 - Moved `to_yaml` method from `Project` to `Task` class
 - converted `Project` class to `project` factory method that returns a `Task` object

### CODE REVIEW

**Comment by capsulecorplab** [comment]
@yamad care to be the (unofficial) reviewer? :grin:

File: mindgraph/graph.py [hunk 1]
```diff
@@ -4,109 +4,109 @@
...
             self._name = name
         else:
             raise TypeError
 
-    def append(self, newnode) -> "Node":
-        """ Creates a new Node and appends it to threads """
-        if type(newnode) is str:
-            newnode = Node(newnode)
-        elif type(newnode) is not Node:
+    def append(self, newTask) -> "Task":
```
Comment (yamad) [inline mindgraph/graph.py]: this looks like a find/replace error. you probably don't want to introduce camelCase. I suggest `newtask` or `new_task`

File: mindgraph/graph.py [hunk 2]
```diff
@@ -4,109 +4,109 @@
...
+    def __getitem__(self, key: int) -> "Task":
+        return self._subtasks[key]
 
     def __repr__(self) -> str:
         return '\n'.join(self.format_tree())
 
-    def format_tree(self: "Node", depth: int = 0) -> Iterator[str]:
-        """Format node and dependents in tree format, emitting lines
+    def format_tree(self: "Task", depth: int = 0) -> Iterator[str]:
+        """Format Task and dependents in tree format, emitting lines
```
Comment (yamad) [inline mindgraph/graph.py]: optional. could clarify this comment "Format Task and subtasks in tree format..."

File: mindgraph/graph.py [hunk 3]
```diff
@@ -4,109 +4,109 @@
...
 
-        Assumes no cycles in graph
+        Assumes no cycles in Project
```
Comment (yamad) [inline mindgraph/graph.py]: Probably needs clarification or removal. a cycle in a graph is realtively clear. what's a cycle in a project?

File: mindgraph/graph.py [hunk 4]
```diff
@@ -4,109 +4,109 @@
...
             yield from n.format_tree(depth+1)
 
     def _postorder(self,
                    depth: int = 0,
-                   visited: Set["Node"] = None,
-                   node_key: Callable[["Node"], Any]=None,
-                   ) -> Generator[Tuple[int, "Node"], None, Set["Node"]]:
-        """Post-order traversal of graph rooted at node"""
+                   visited: Set["Task"] = None,
+                   Task_key: Callable[["Task"], Any]=None,
```
Comment (yamad) [inline mindgraph/graph.py]: another find/replace error. `Task_key` --> `task_key` (and I should probably find a clearer name (or at least document it's purpose) in the future.

File: mindgraph/graph.py [hunk 5]
```diff
@@ -4,109 +4,109 @@
...
 
-        Nodes are scheduled by weight and to resolve blocking tasks
+        Tasks are scheduled by priority and to resolve blocking tasks
         """
-        # sorts by weight (2 before 1), then alphabetical
-        def node_key(node):
-            return (-node.weight, node.name)
-        return (x[1] for x in self._postorder(node_key=node_key))
+        # sorts by priority (2 before 1), then alphabetical
+        def Task_key(Task):
```
Comment (yamad) [inline mindgraph/graph.py]: more capitalization issues

File: test/test_mindgraph.py [hunk 6]
```diff
@@ -39,13 +39,13 @@ def task_graph():
...
-    t12.threads.append(t22)
+    t32.subtasks.append(t22)
+    t12.subtasks.append(t22)
     return g
 
 
-def test_todo_high_weights_win(task_graph):
-    """High weights are scheduled before low weights"""
+def test_todo_high_prioritys_win(task_graph):
+    """High prioritys are scheduled before low prioritys"""
```
Comment (yamad) [inline test/test_mindgraph.py]: `prioritys` --> `priorities`

File: test/test_mindgraph.py [hunk 7]
```diff
@@ -60,24 +60,24 @@ def test_todo_blocking_tasks_win(task_graph):
     assert todo.index('task 1.1') < todo.index('task 1.2')
 
 
-def test_postorder_default_weights_ignored(task_graph):
-    """Post-order traversal ignores node weights by default"""
+def test_postorder_default_prioritys_ignored(task_graph):
+    """Post-order traversal ignores node prioritys by default"""
```
Comment (yamad) [inline test/test_mindgraph.py]: more priorities

**Review by yamad — CHANGES_REQUESTED** [review]
Overall, the names clarify the intent so much. Really good change.

Just a few details to correct from the batch substitutions.

File: mindgraph/graph.py (hunk 1 above)
Comment (capsulecorplab) [reply mindgraph/graph.py]: fixed in 6cae8b6!

File: mindgraph/graph.py (hunk 5 above)
Comment (capsulecorplab) [reply mindgraph/graph.py]: memo: i need to stop using "replace all"...

File: mindgraph/graph.py (hunk 4 above)
Comment (capsulecorplab) [reply mindgraph/graph.py]: fixed in 77b8f69

File: mindgraph/graph.py (hunk 5 above)
Comment (capsulecorplab) [reply mindgraph/graph.py]: fixed in 77b8f69

File: mindgraph/graph.py (hunk 2 above)
Comment (capsulecorplab) [reply mindgraph/graph.py]: went w/ "Generates task and subtasks into a string formatted tree" :man_shrugging:

File: mindgraph/graph.py (hunk 3 above)
Comment (capsulecorplab) [reply mindgraph/graph.py]: went w/ removal.

**Pull request merged**